unicorn_contracts/src/contracts_service/createContractFunction.ts

```diff
@@ -32,96 +24,92 @@ class CreateContractFunction implements LambdaInterface {
-                'contract_id': contractId,
-                'contract_last_modified_on': dbEntry.contract_created?? '',
-                'contract_status': dbEntry.contract_status,
-                'property_id': dbEntry.property_id?? ''
+        // Construct the DDB Table record 
+        logger.info('Constructing DB Entry from data', { data });
+        const dbEntry: ContractDBType = {
+            property_id: data['property_id'],
+            contract_created: createDate.toISOString(),
+            contract_last_modified_on: createDate.toISOString(),
+            contract_id: contractId,
+            address: data['address'],
+            seller_name: data['seller_name'],
+            contract_status: ContractStatusEnum.DRAFT
+        };
+
+        try {
+            // Save the entry.
+            await this.createContract(dbEntry, contractId);
+            tracer.putAnnotation('ContractStatus', JSON.stringify(dbEntry));
+        } catch (error) {
+            tracer.addErrorAsMetadata(error as Error);
+            logger.error('Error during DDB PUT', error as Error);
+
+            return {
+                statusCode: 500,
+                body: JSON.stringify({error: `Unable to create contract ${contractId}`})
             };
+        }
+
+        // Fire off an event for 'Contract created' 
+        const contractStatusChanged: ContractStatusChangedEvent = {
+            'contract_id': contractId,
+            'contract_last_modified_on': dbEntry.contract_created?? '',
+            'contract_status': dbEntry.contract_status,
+            'property_id': dbEntry.property_id?? ''
+        };
 
-            try {
-                await this.publishEvent(contractStatusChanged, contractId);
-            } catch (error: any) {
-                tracer.addErrorAsMetadata(error as Error);
-                logger.error(`Error during EventBridge PUT: ${JSON.stringify(error)}`);
-                return {
-                    statusCode: 500,
-                    body: JSON.stringify({error: `Unable to fire 'Contract created' event for contract ${contractId}`})
-                };
-            }
-
-            const apiResponse: APIGatewayProxyResult = {
-                statusCode: 200,
-                body: JSON.stringify(contractStatusChanged)
+        try {
+            await this.publishEvent(contractStatusChanged, contractId);
+        } catch (error) {
+            tracer.addErrorAsMetadata(error as Error);
+            logger.error('Error during EventBridge PUT:', error as Error);
+
+            return {
+                statusCode: 500,
+                body: JSON.stringify({error: `Unable to fire 'Contract created' event for contract ${contractId}`})
             };
-            tracer.addResponseAsMetadata(apiResponse, process.env._HANDLER);
-            return apiResponse;
-        } finally {
-            // Don't forget to publish your metrics.
-            metrics.publishStoredMetrics();
```

Comment:
You don't need to do this, the `logMetrics` decorator already does this for you at the end of the method regardless of whether you throw or return.